Features/DetailView/MovieDetailViewController.swift

```diff
@@ -106,22 +126,22 @@ private extension MovieDetailViewController {
     }
     
     @objc func dismissView(_ sender: UIButton) {
+        delegate?.updateView(self)
+        
         dismiss(animated: true)
     }
 }
 
 // MARK: - WatchlistButtonDelegate
 extension MovieDetailViewController: WatchlistButtonDelegate {
     func watchlistTapped(_ view: WatchlistHandleable) {
-        guard let movieDetails = movieDetails else { return }
-        
-        let watchlistItem = WatchlistItem(id: selectedMovieID,
-                                          saveDate: Date.now,
-                                          title: movieDetails.title,
-                                          voteAverage: movieDetails.voteAverage,
-                                          posterPath: movieDetails.posterPath)
+        guard let movie = movieDetails else { return }
 
-        let updatedStatus = watchlistService.toggleStatus(for: watchlistItem)
+        let updatedStatus = watchlistService.toggleStatus(for: WatchlistMovieConfiguration(id: Int32(movie.id),
+                                                                                           saveDate: Date.now,
+                                                                                           title: movie.title,
+                                                                                           voteAverage: Int16(movie.voteAverage),
+                                                                                           posterPath: movie.posterPath))
```

Comment:
Since `toggleStatus(for item: WatchlistMovieConfiguration)` function has to accept not only `MovieDetails`, but also `Movie` instances, I've created `CoreDataHandleable` protocol (not sure about naming as usual) to make one initializer for both and for possible new data models. And a separate initializer for `WatchlistMovie` instances.